docs/reference/api/platform/FileHandle.md

```diff
@@ -0,0 +1,274 @@
+## FileHandle
+
+For general information on [files](file.html) and [filing systems](filesystem.html), see the respective documentation. This chapter covers the abstract API, with an emphasis on devices, as they're less straightforward.
+
+`FileHandle` is an abstract class representing a device that supports file-like operations such as `read` and `write`. This may be an actual `File` on a storage device provided by a `FileSystem`, or a device such as `UARTSerial`.
+
+The `FileHandle` abstraction represents an already-opened file or device, so it has no `open` method of its own - the opening may take the form of a call to another class that returns a `FileHandle`, such as `FileSystem::open`, or it may be implicit in the construction of an object such as `UARTSerial`. 
+
+The `FileHandle` abstraction permits stream-handling code to be device-independent. For example the console input and output streams used for C's `stdin` and `stdout` can be retargeted to something other than the default serial port via the `FileHandle` API, and `ATCmdParser` and the PPP connection to lwIP work on abstract `FileHandle` pointers.
+
+Exactly which operations a `FileHandle` supports will depend on the underlying device, and will in turn restrict what applications it is suitable for. For example, a database application might require random-access and `seek`, but this may not be available on a limited filesystem, and certainly not on a stream device. Only a `File` on a full `FileSystem` such as `FATFileSystem` would generally implement the entire API. Specialised devices may have particular limitations or behavior, which they should document, and which would limit their general utility.
```

Comment:
Shoud we note that if not supported, implementations should return ENOSYS?

Reply:
Will note that errors should be indicated - sometimes there's a more specific code, eg `ESPIPE` for `invalid seek`.

Reply:
Good point